src/server/svr_func.c

```diff
@@ -4931,15 +4931,23 @@ find_prov_vnode_list(job *pjob, exec_vnode_listtype *prov_vnodes, char **aoe_nam
 					DBPRT(("%s: %s\n", __func__, (*prov_vnodes)[i]))
 					++i;
 					if (aoe_name != NULL) {
-						aoe = malloc(strlen(((pkvp + k)->kv_val)) + 1);
-						if (aoe == NULL) {
-							free(sbuf);
-							free(pbuf);
-							return -1;
+						if (*aoe_name) {
+							if (strcmp(*aoe_name, ((pkvp + k)->kv_val)) != 0) {
```

Comment:
this should already be caught by the action function resc_select_action() right? I don't think we need this check here